tests/unit-tests/product/data-store.php

```diff
@@ -910,4 +913,52 @@ public function test_variable_create_all_product_variations_limits() {
 		$count = $data_store->create_all_product_variations( wc_get_product( $product_id ), 10 );
 		$this->assertEquals( 7, $count );
 	}
+
+	/**
+	 * Test find_matching_product_variation.
+	 *
+	 * @return void
+	 */
+	public function test_find_matching_product_variation() {
+		$product    = WC_Helper_Product::create_variation_product();
+		$data_store = WC_Data_Store::load( 'product' );
+		$children   = $product->get_children();
+
+		$match = $data_store->find_matching_product_variation( $product, array() );
+		$this->assertEquals( 0, $match );
+
+		$match = $data_store->find_matching_product_variation(
+			$product,
+			array(
+				'attribute_pa_size' => 'small',
+			)
+		);
+		$this->assertEquals( $children[0], $match );
+
+		$match = $data_store->find_matching_product_variation(
+			$product,
+			array(
+				'attribute_pa_size' => 'large',
+			)
+		);
+		$this->assertEquals( $children[1], $match );
+
+		$match = $data_store->find_matching_product_variation(
+			$product,
+			array(
+				'attribute_pa_size'   => 'small',
+				'attribute_pa_colour' => '',
+			)
+		);
+		$this->assertEquals( $children[0], $match );
+
+		$match = $data_store->find_matching_product_variation(
+			$product,
+			array(
+				'attribute_pa_size'   => 'small',
+				'attribute_pa_colour' => 'red',
+			)
+		);
+		$this->assertEquals( $children[0], $match );
```

Comment:
Great that this change comes in with unit tests! 
Should this test also explore other eventualities, like if only the colour is defined or if non-existing values are passed to the function?